### ISSUE TITLE
Compatibility with Terraform 1.10

### ISSUE THREAD

**Issue by djacquensf9** [gruntwork-io/terragrunt#3604]
Hi,

## Describe the enhancement

Terraform 1.10 was released two hours ago and Terragrunt is not yet compatible with this version !! 

Two hours !! What are you doing, guys ?! 🤣 

Thanks, ;)

David

**Comment by denis256** [comment]
Hi,
the new Terraform 1.10 introduces changes in input/plan handling and S3 configuration behavior, which have caused multiple Terragrunt tests to fail.

If you'd like to contribute fixes or insights, we’d greatly appreciate your support. Contributions from the community play a significant role in making updates faster and smoother.

Example failing tests:
TestAutoRetryGetDefaultErrors
TestTerragruntWorksWithIncludeDeepMerge
TestTerragruntWorksWithMultipleInclude
TestSopsDecryptedCorrectly
TestDependencyMockOutputMergeStrategyWithStateCompatTrue
TestDependencyMockOutputMergeStrategyWithStateDeepMapOnly
TestDependencyOutputTypeConversion
TestTerragruntRunAllPlanAndShow
TestAwsSopsDecryptedKMSCorrectly


Reference PR
https://github.com/gruntwork-io/terragrunt/pull/3605

**Comment by ricardoleal** [comment]
For now i'm getting the following

```
  ╷
  │ Error: Can't change variable when applying a saved plan
  │ 
  │ The variable asg_capacity cannot be set using the -var and -var-file
  │ options when applying a saved plan file, because a saved plan includes the
  │ variable values that were set when it was created. The saved plan specifies
  │ "{\"desired_capacity\":1,\"max_size\":1,\"min_size\":1}" as the value
  │ whereas during apply the value object with 3 attributes was set by an
  │ environment variable. To declare an ephemeral variable which is not saved
  │ in the plan file, use ephemeral = true.
  ╵
```

**Comment by joshbsemperis** [comment]
Experiencing the same, seems like quite a serious issue, its a breaking change

**Comment by ricardoleal** [comment]
> For now i'm getting the following
> 
> ```shell
>   ╷
>   │ Error: Can't change variable when applying a saved plan
>   │ 
>   │ The variable asg_capacity cannot be set using the -var and -var-file
>   │ options when applying a saved plan file, because a saved plan includes the
>   │ variable values that were set when it was created. The saved plan specifies
>   │ "{\"desired_capacity\":1,\"max_size\":1,\"min_size\":1}" as the value
>   │ whereas during apply the value object with 3 attributes was set by an
>   │ environment variable. To declare an ephemeral variable which is not saved
>   │ in the plan file, use ephemeral = true.
>   ╵
> ```
> 
> ```shell
> ➜  terragrunt --version
> terragrunt version v0.69.2
> ➜  terraform --version 
> Terraform v1.10.0
> on darwin_arm64
> + provider registry.terraform.io/hashicorp/aws v5.78.0
> + provider registry.terraform.io/hashicorp/cloudinit v2.3.5
> ```
> 
> Update with `terraform` and `terragrunt` versions
> 
> Issue on hashicorp for reference : [hashicorp/terraform#36106](https://github.com/hashicorp/terraform/issues/36106)

Issue solved.

Tested new backend with s3 state lock and working.

```hcl 
remote_state {
  backend = "s3"
  config = {
    bucket         = "xxxxx-bucket"
    key            = format("%s/terraform.tfstate", get_path_from_repo_root())
    region         = "eu-west-3"
    encrypt        = true
    dynamodb_table = "xxxxxx-db-table"
    use_lockfile   = true
  }
}
```

result
```json
aws s3 cp s3://xxxxx-bucket/terraform.tfstate.tflock - | jq .
{
  "ID": "ee0b3c54-de6a-d4ca-d0a6-52f5298ed7c9",
  "Operation": "OperationTypeApply",
  "Info": "",
  "Who": "xxxx@device",
  "Version": "1.10.1",
  "Created": "2024-12-04T16:54:05.570709Z",
  "Path": "xxxxx-bucket/terraform.tfstate"
}
```

**Comment by liangrog** [comment]
Terraform has released 1.10.1 for the fix. Could we bump the version up to the same for Terragrunt docker container? Thanks.

**Comment by yhakbar** [comment]
Hey folks,

We're still seeing failures in our CI testing the integration with 1.10.1 related to applying stored plans (the three failing tests are `TestTerragruntRunAllPlanAndShow`, `TestStorePlanFilesRunAllPlanApplyRelativePath` and `TestStorePlanFilesRunAllPlanApply` if interested).

We won't bump the officially supported version until it passes CI.

**Comment by yhakbar** [comment]
The recent release of 1.10.2 seems to have a release note related to fixing applying plans, so I'm re-running integration tests on that version.

**Comment by yhakbar** [comment]
Still seeing failures in CI:

TestStorePlanFilesRunAllPlanApply:
```
        	            	* Failed to execute "terraform apply -auto-approve -input=false /tmp/TestStorePlanFilesRunAllPlanApply3764107339/001/app/tfplan.tfplan" in /tmp/terragrunt-test857549231/fixtures/out-dir/app
        	            	
        	            	  Error: Can't change variable when applying a saved plan
        	            	
        	            	  The variable input_value cannot be set using the -var and -var-file options
        	            	  when applying a saved plan file, because a saved plan includes the variable
        	            	  values that were set when it was created. The saved plan specifies "42" as
        	            	  the value whereas during apply the value "46521694" was set by an
        	            	  environment variable. To declare an ephemeral variable which is not saved
        	            	  in the plan file, use ephemeral = true.
```

**Comment by alandreasyans** [comment]
Hi all,
Testing now the new feature of the native state locking with a terragrunt, however, getting issues.
Terraform version -1.10.5
Terragrunt version - v0.73.0-beta2025020601(was same with a 0.72.0 version)

My backend configuration.

`remote_state {
  generate = {
    path      = "backend.tf"
    if_exists = "overwrite"
  }
  backend = "s3"
  config = {
    encrypt          = true
    bucket           = "my-bucket-${get_aws_account_id()}"
    key                = local.key
    region            = "eu-west-2"
    use_lockfile   = true
  }
}
`

Tried to reconfigure the state or migrate state arguments, but getting the same issue.
`error: Backend initialization required, please run "terraform init"
│ 
│ Reason: Initial configuration of the requested backend "s3"
│ 
│ The "backend" is the interface that Terraform uses to store state,
│ perform operations, etc. If this message is showing up, it means that the
│ Terraform configuration you're using is using a custom configuration for
│ the Terraform backend.
│ 
│ Changes to backend configurations require reinitialization. This allows
│ Terraform to set up the new configuration, copy existing state, etc. Please
│ run
│ "terraform init" with either the "-reconfigure" or "-migrate-state" flags
│ to
│ use the current configuration.
│ 
│ If the change reason above is incorrect, please verify your configuration
│ hasn't changed and try again. At this point, no changes to your existing
│ configuration or state have been made.`

**Comment by tropnikovvl** [comment]
Hi, this functionality would be nice

**Comment by sassdavid** [comment]
In the meantime Terraform 1.11.0 has been release, we hope it could help to move ahead with the compatibility of the last two versions.

**Comment by djacquensf9** [comment]
> In the meantime Terraform 1.11.0 has been release, we hope it could help to move ahead with the compatibility of the last two versions.

Hi everyone, 😉 

Thank you very much everyone for your great feedbacks.

As @sassdavid pointed out, 1.11.0 has been released. 
Could someone create a new issue for 1.11.0 so I can close this issue? 

Thanks 🌻 

David

**Comment by wanis-fahmy** [comment]
Hey folks, thank you for the update! does it take time till the release is reflected in brew? https://formulae.brew.sh/api/formula/terragrunt.json still points to the previous release

**Comment by yhakbar** [comment]
@wanis-fahmy , yup! Once this is merged, homebrew will reference `v0.74.0`:
https://github.com/Homebrew/homebrew-core/pull/209572
